code/go/0chain.net/blobbercore/writemarker/worker.go

```diff
@@ -16,52 +16,80 @@ func SetupWorkers(ctx context.Context) {
 	go startRedeemWriteMarkers(ctx)
 }
 
-func RedeemMarkersForAllocation(ctx context.Context, allocationObj *allocation.Allocation) error {
-	rctx := datastore.GetStore().CreateTransaction(ctx)
-	db := datastore.GetStore().GetTransaction(rctx)
+func redeemWriterMarkersForAllocation(allocationObj *allocation.Allocation) {
 	defer func() {
-		err := db.Commit().Error
-		if err != nil {
-			Logger.Error("Error committing the writemarker redeem", zap.Error(err))
+		if r := recover(); r != nil {
```

Comment:
I think if there is any panicking code, then we should find it and solve it because this will be blocking blobber from redeeming write markers.
Above `defer` will only skip panic and log the error. My point is, there should be no any panicking code while redeeming writemarkers.

Reply:
```
writemarker/writemarker.go:17 [recover] redeemWriteMarker {"err": "runtime error: invalid memory address or nil pointer dereference"}
github.com/0chain/blobber/code/go/0chain.net/blobbercore/writemarker.redeemWriteMarker.func1
/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/writemarker/writemarker.go:17
runtime.gopanic
/usr/local/go/src/runtime/panic.go:838
runtime.panicmem
/usr/local/go/src/runtime/panic.go:220
runtime.sigpanic
/usr/local/go/src/runtime/signal_unix.go:818
database/sql.(*Tx).rollback
/usr/local/go/src/database/sql/sql.go:2304
database/sql.(*Tx).Rollback
/usr/local/go/src/database/sql/sql.go:2336
gorm.io/gorm.(*PreparedStmtTX).Rollback
/Users/geax/code/go/pkg/mod/gorm.io/gorm@v1.23.3/prepare_stmt.go:131
gorm.io/gorm.(*DB).Rollback
/Users/geax/code/go/pkg/mod/gorm.io/gorm@v1.23.3/finisher_api.go:629
github.com/0chain/blobber/code/go/0chain.net/blobbercore/writemarker.redeemWriteMarker
/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/writemarker/writemarker.go:40
github.com/0chain/blobber/code/go/0chain.net/blobbercore/writemarker.startRedeemWriteMarkers
/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/writemarker/worker.go:78
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x44e02c2]
```

Reply:
anyway, removed `recover` code as per suggestion

Reply:
but i think it is still possible to panic if network has any issue.